packages/react-components/react-tree/src/hooks/useHTMLElementWalker.ts

```diff
@@ -0,0 +1,101 @@
+import { isHTMLElement } from '@fluentui/react-utilities';
+import * as React from 'react';
+import { assertIsDefinedRef } from '../utils/assert';
+
+export interface HTMLElementWalker {
+  readonly root: HTMLElement;
+  currentElement: HTMLElement;
+  firstChild(filter?: HTMLElementFilter): HTMLElement | null;
+  lastChild(filter?: HTMLElementFilter): HTMLElement | null;
+  nextElement(filter?: HTMLElementFilter): HTMLElement | null;
+  nextSibling(filter?: HTMLElementFilter): HTMLElement | null;
+  parentElement(filter?: HTMLElementFilter): HTMLElement | null;
+  previousElement(filter?: HTMLElementFilter): HTMLElement | null;
+  previousSibling(filter?: HTMLElementFilter): HTMLElement | null;
+}
+
+export type HTMLElementFilter = (element: HTMLElement) => number;
+
+const rootRefErrorMessage = 'useHTMLElementWalker: root reference not properly provided';
+
+export function useHTMLElementWalker(filter?: HTMLElementFilter) {
+  const rootRef = React.useRef<HTMLElement>(null);
+  const treeWalkerRef = React.useRef<TreeWalker>();
+  const temporaryFilterRef = React.useRef<HTMLElementFilter>();
+  React.useEffect(() => {
+    assertIsDefinedRef(rootRef, rootRefErrorMessage);
+    treeWalkerRef.current = document.createTreeWalker(rootRef.current, NodeFilter.SHOW_ELEMENT, {
+      acceptNode(node: Node) {
+        if (!isHTMLElement(node)) {
+          return NodeFilter.FILTER_REJECT;
+        }
+        const filterResult = filter?.(node) ?? NodeFilter.FILTER_ACCEPT;
+        return filterResult === NodeFilter.FILTER_ACCEPT
+          ? temporaryFilterRef.current?.(node) ?? filterResult
+          : filterResult;
+      },
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why are you disabling dependencies here?

Reply:
poorly implement, when using callback ref this is unnecessary.